drivers/hmis/app/models/hmis/ce/opportunity_refresher.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+module Hmis::Ce
+  class OpportunityRefresher
+    # Close stale opportunities and create fresh ones with updated CE Match Rules
+    #
+    # @param candidate_pool_ids [Array<Integer>, nil] Optional array of candidate pool IDs to filter by.
+    #   If nil, processes ALL stale opportunities across the app.
+    # @return [Hash] {
+    #   closed_count: Integer,
+    #   closed_opportunity_unit_ids: Array<Integer>,
+    #   created_count: Integer,
+    #   created_opportunity_ids: Array<Integer>,
+    # }
+    def refresh_stale_opportunities(candidate_pool_ids: nil)
+      scope = Hmis::Ce::Opportunity.stale.active.open
```

Comment:
```suggestion
      scope = Hmis::Ce::Opportunity.stale.open
```
Can remove redundant scope (`where.not(status: 'closed').where(status: 'open')`)